### PR TITLE
New trail form

### PR DESCRIPTION
**Pull request by benjamin-z-l**
Added a new trail form with back end validations.

### CODE REVIEW

File: src/main/java/com/launchacademy/reviews/controllers/TrailsApiV1Controller.java [hunk 1]
```diff
@@ -36,4 +43,16 @@ public TrailsApiV1Controller(TrailService trailService) {
...
+
+  @PostMapping
+  public ResponseEntity<Trail> addNewTrail(@RequestBody @Valid Trail trail, BindingResult bindingResult) {
+    if (bindingResult.hasErrors()) {
+      System.out.println("ERRORS");
+      throw new InvalidFormDataException(bindingResult.getFieldErrors());
+    }
+    else {
+      System.out.println("Hello");
+      return new ResponseEntity<>(trailService.save(trail), HttpStatus.OK);
```
Comment (safang99) [inline src/main/java/com/launchacademy/reviews/controllers/TrailsApiV1Controller.java:53]: Let's remove the sout statements on 50 and 54 now that this feature has been tested and is working.

File: src/main/java/com/launchacademy/reviews/exceptions/ControllerExceptionHandler.java [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
+  @ResponseBody
+  public Map<String, Map<String, String>> InvalidFormDataExceptionHandler(InvalidFormDataException e) {
+    Map<String, String> fieldErrors = new HashMap<>();
+    for ( FieldError fieldError : e.getFieldErrors()) {
+      fieldErrors.put(fieldError.getField(), fieldError.getDefaultMessage());
+    }
+    Map<String, Map<String, String>> errors = new HashMap<>();
+    errors.put("errors", fieldErrors);
+    return errors;
+  }
```
Comment (safang99) [inline src/main/java/com/launchacademy/reviews/exceptions/ControllerExceptionHandler.java:26]: Very nice!

**Review by safang99 — CHANGES_REQUESTED** [review]
Awesome work here! Just two small changes and then feel free to merge on your own.  You can also repost the PR after making the changes in the questions channel and I can merge it when I get there too.

File: src/main/java/com/launchacademy/reviews/controllers/TrailsApiV1Controller.java (hunk 1 above)
Comment (benjamin-z-l) [reply src/main/java/com/launchacademy/reviews/controllers/TrailsApiV1Controller.java:53]: Removed!

**Pull request merged**